.github/actions/python_prepare/action.yml

```diff
@@ -0,0 +1,48 @@
+name: "Prepare development environment"
+description: "Set up Python and Node.js environments with caching"
+runs:
+  using: "composite"
+  steps:
+    # Python setup
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.10.14
+    
+    - name: Cache Poetry installation
```

Comment:
Nice idea with cache!

I didn't use that in Github before,[ but from their repo](https://github.com/actions/cache?tab=readme-ov-file#restoring-and-saving-cache-using-a-single-action), it seems like there needs to be something as 

```
if: steps.cache-primes.outputs.cache-hit != 'true'
```

in the follow-up step. Or did you test that Poetry will quickly skip the installation by itself?

Reply:
Please see latest changes and the corresponding CI run -> https://github.com/gnosis/python-web3-wallet-streamlit/actions/runs/12692379504?pr=6

I believe the `cache` is working now 🙏